### DISCUSSION TITLE
Thermostat Mode : Add Telemetry output - asking for info to implement

### DISCUSSION THREAD

**Discussion by FulvioSpelta** [arendst/Tasmota#13402] [Q&A] [answered]
PROBLEM DESCRIPTION
I'd like to add a telemetry json (eg. like the SENSOR one) for Thermostat mode, in order to let easier to integrate if with external system (eg. openhab).

REQUESTED INFORMATION
I can try to write the code requested to implement it and propose a PR or, at least, use in my own implementation :-)

I kindly ask for comments or ideas and few information in order to quickly be oriented in the code and how to implement it following the code style.

Happy to share more info if needed.
Thanks

**Comment by ascillato** [comment, marked as answer]
Hi,

That is already implemented.

Please, check the Thermostat driver at https://github.com/arendst/Tasmota/blob/development/tasmota/xdrv_39_thermostat.ino

Thanks.

**Reply by FulvioSpelta** [reply]
Thanks for the prompt reply, could you be so kind to give more info about the code where it's implemented ?

I've searched per "publish" in the code (before post here too)  but the only 2 calls to "MqttPublish" are in debug code and searching for "tele" found nothing.

Thanks a lot

**Reply by FulvioSpelta** [reply]
To be clearer: i'd like to have a json with thermostat own information sech as TempTargetSet, TempMeasuredSet, controllermodeset and some interval parameter of regulation algorithms.

**Reply by ascillato** [reply]
Yes, it is exactly that. The driver is in the link I have posted to you. It is already part of Tasmota codebase.

For more information on the usage of that driver, please check Tasmota docs at: https://tasmota.github.io/docs/Thermostat/

**Reply by FulvioSpelta** [reply]
I've read both the code and the documentation and I've implemented a thermostat on a sonoff 4ch R3 using 2 ds18b20 sensors that is running properly but i'm not able to find or activate the termostat telemetry.

Under tele topic i get only

- LWT
- STATE
- SENSOR

Sorry but I need more specific info.

In my implementation i've created rules to publish the info i need to integrate in openhab.
